### PR TITLE
Extract the transcription retry and resume workflow

### PR DESCRIPTION
**Pull request by woosublee**
## Summary

- add an instance-owned `TranscriptionRetryWorkflow` for Manual Retry and durable startup Resume, using the existing execution snapshots, cloud sidecars, history access, and note asset boundaries
- keep `AppState` as the UI-facing adapter that captures current Manual policy, restores persisted startup policy and context, mirrors workflow state, performs interactive delivery only for Manual Retry, and invalidates Meeting Summary only after durable Manual results
- reject stale or cancelled completions, preserve unrelated note metadata, keep history-before-sidecar cleanup ordering explicit, and retain runtime-only credentials outside durable state and diagnostics
- replace embedded Retry/Resume source-shape assertions with direct workflow behavior coverage plus narrow AppState integration contracts
- fix review-confirmed lifecycle regressions around startup event wiring, whitespace-sensitive resumed output, replaced cloud progress, duplicate terminal cancellation outcomes, duplicate history IDs, and non-blocking test synchronization

## Verification

- `make check-test-wiring`
- `make test`
  - core: 55s
  - recording: 82s
  - transcription: 121s
- signed production build
- `codesign --verify --deep --strict`
- `/code-review medium` (run exactly once after the full tree was green; four confirmed regressions were fixed and the full verification was repeated)
- follow-up CodeRabbit findings were verified, covered by regression tests or stronger source contracts, and the full CI suite passed on the final commit

Closes #318

## Summary by CodeRabbit

- **새로운 기능**
  - 실패한 클라우드 전사를 수동으로 다시 시도할 수 있습니다.
  - 앱 시작 후 중단된 전사 작업을 자동으로 재개합니다.
  - 전사 진행 상태와 성공·대체 처리·실패·취소 결과를 안정적으로 반영합니다.
  - 재시도 시 기존 오디오, 언어, 문맥 및 처리 설정을 유지합니다.

- **개선 사항**
  - 전사 편집, 삭제 또는 초기화 시 진행 중인 재시도를 안전하게 취소합니다.
  - 재시도 결과와 오류 상태를 기록에 저장하고, 관련 경고와 요약 정보를 올바르게 갱신합니다.

### CODE REVIEW

File: Sources/AppState.swift [hunk 1]
```diff
@@ -4940,6 +4923,73 @@ final class AppState: ObservableObject, @unchecked Sendable {
...
+                        providerDetail: cleanupFailureDescription
+                    )
+                }
+
+            case .persistenceFailed(let issue):
+                errorMessage = issue.presentation().compactMessage
+
+            case .failed, .cancelled, .stale:
+                break
+            }
```
[the remark this answers is not in the document]
Comment (woosublee) [reply Sources/AppState.swift:4969]: Thanks, @coderabbitai. I verified this against the pre-extraction behavior on origin/main: Manual Retry failures were persisted on the history item but did not also set the menu-bar errorMessage. Changing that here would introduce new user-visible copy and presentation behavior rather than preserve the workflow during extraction, so I am leaving it unchanged in this focused PR.

File: Sources/AppState.swift [hunk 2]
```diff
@@ -10468,184 +10496,52 @@ final class AppState: ObservableObject, @unchecked Sendable {
...
-              let audioFileName = item.audioFileName else {
-            return
-        }
-        let audioURL = storageLayout.audioDirectory.appendingPathComponent(
-            audioFileName
-        )
-        retryingItemIDs.insert(item.id)
-        let session = cloudTranscriptionJobStore.beginSession(
-            historyID: record.historyID
         )
```
[the remark this answers is not in the document]
Comment (woosublee) [reply Sources/AppState.swift:10541]: Thanks, @coderabbitai. I checked the previous startup path and the approved policy for this extraction: startup Resume intentionally restores the stored context and persisted completion policy, while only Manual Retry uses the current context toggle. Applying disableContextCapture here would change durable resume semantics, so I am leaving the startup behavior unchanged in this focused PR.

File: Sources/TranscriptionRetryWorkflow.swift [hunk 3]
```diff
@@ -0,0 +1,1015 @@
...
+        let reconciliation = CloudTranscriptionStartupReconciler(
+            store: runtime.cloud.jobStore,
+            audioRoot: input.audioDirectory
+        ).reconcile(
+            input.reconciliation,
+            runtime: input.runtime
+        )
+        let historyByID = Dictionary(
+            uniqueKeysWithValues: input.history.map { ($0.id, $0) }
+        )
```
[the remark this answers is not in the document]
Comment (woosublee) [reply Sources/TranscriptionRetryWorkflow.swift:336]: Fixed in 5375c2b. Startup history indexing now uses uniquingKeysWith to retain the first item, matching the previous first(where:) behavior without trapping on duplicate IDs. I also added a direct workflow regression test that supplies duplicate IDs and verifies the first stored context is used.

File: Tests/AppStateTranscriptionConfigurationTests.swift [hunk 4]
```diff
@@ -2,6 +2,108 @@ import Combine
...
+        condition.lock()
+        defer { condition.unlock() }
+        let deadline = Date().addingTimeInterval(3)
+        while !didFinishUpload {
+            precondition(
+                condition.wait(until: deadline),
+                "Timed out waiting for controlled Retry upload to finish"
+            )
+        }
+    }
```
[the remark this answers is not in the document]
Comment (woosublee) [reply Tests/AppStateTranscriptionConfigurationTests.swift]: Fixed in 5375c2b. The tests now poll the lock-protected state asynchronously instead of blocking a cooperative thread with NSCondition.wait, and the controlled upload stores an array of continuations so overlapping waiters are all resumed.

File: Tests/QuillUserIssueUIContractTests.swift [hunk 5]
```diff
@@ -238,14 +238,15 @@ struct QuillUserIssueUIContractTests {
...
-            to: "private func copyRetryTranscriptToPasteboardIfNeeded"
+            from: "private func applyTranscriptionRetryWorkflowEvent(",
+            to: "private func applyTranscriptionRetryWorkflowState("
         )
         try expect(
-            retryBody.contains("incrementNoteRetryGeneration(for: snapshot.item.id)"),
-            "saved retry results invalidate stale dismissals"
+            retryEvents.contains("if effects.advancesWarningGeneration")
+                && retryEvents.contains("incrementNoteRetryGeneration(for: item.id)"),
+            "persisted Retry effects invalidate stale dismissals"
```
[the remark this answers is not in the document]
Comment (woosublee) [reply Tests/QuillUserIssueUIContractTests.swift]: Fixed in 5375c2b. The source contract now extracts the advancesWarningGeneration conditional block and asserts that incrementNoteRetryGeneration(for:) is nested inside that block, rather than checking the two markers independently.
